### PR TITLE
show total count for drafts folder

### PR DESCRIPTION
**Pull request by tahaalibra**
this solves issue #540 

Currently the drafts shows unseen/unread count. this solution would enable to show total count for the drafts folder only.

review @Gomez @ChristophWurst, This is my first PR, i am open for code review

### CODE REVIEW

File: js/templates/folder.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 	<a class="folder {{#if specialRole}} icon-{{specialRole}} svg{{/if}}">
 		{{name}}
 		{{#if unseen}}
```
Comment (Gomez) [inline js/templates/folder.html]: We go in only if unseen. Drafts are always unseen. So you need to do further tests here.

File: js/views/helper.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ define(function(require) {
 		return t('mail', text);
 	});
 
+	Handlebars.registerHelper('printTotalOrUnseenCount', function(folder, unseen, total) {
+		if(folder == 'drafts') {
+			return total;
```
Comment (Gomez) [inline js/views/helper.js]: jscs reports here: "requireSpaceAfterKeywords: One space required after "if" keyword at". You should install jscs and test your style.

**Comment by Gomez** [comment]
Thanks for this PR! Please add a changelog enty, too.

**Comment by ChristophWurst** [comment]
@tahaalibra thanks for that PR. I'd prefer to have that view logic inside the [folder view](https://github.com/owncloud/mail/blob/master/js/views/folder.js). That can be achieved easily by using [template helpers](https://github.com/marionettejs/backbone.marionette/blob/master/docs/marionette.view.md#viewtemplatehelpers) like we did [here](https://github.com/owncloud/mail/blob/master/js/views/composer.js#L25-L49).

Using a view helper is a valid approach, but it makes the code harder to maintain as you'd have the logic split into two separate locations. There's only one template that would make use of it.

So simply move the filter code over as template helper and fix the bug @Gomez mentioned, thanks!

**Comment by tahaalibra** [comment]
Updated the code...Using template helpers seems much better option

**Comment by ChristophWurst** [comment]
Thanks a lot, that looks very good!

We'll have to decrement that 'total' number when the user deletes a message. Otherwise the counter stays unchanged while deleting drafts. There's code that updates the 'unseen' value, so we just have to adjust that to also update the other value.
@tahaalibra we can either do that in this PR or a follow-up PR. Do you want to have a look at that?

**Comment by ChristophWurst** [comment]
[This](https://github.com/owncloud/mail/blob/master/js/views/messagesitem.js#L82-L135) is where clicks on the trash icon are handled. You can use ``require('state').currentFolder`` to get a reference to the folder model, where the 'total' counter has to be decreased. Changes to that property *should* trigger a re-render of the folder view, but I'm not a 100% sure about that. One has to try ;-)

**Comment by tahaalibra** [comment]
i think this also need to be updated for another case i.e if a user receives a new message...

**Comment by tahaalibra** [comment]
Also the title bar still shows drafts with unseen....i will try to correct these problem...i will contact if i need any help

**Comment by tahaalibra** [comment]
[x] the count on decreases if we delete a message in the drafts folder
[x] the title of the page now shows total count. this is for drafts folder only

one thing though, i checked this without this PR, the unseen count for the index folder does not if suppose the user has loaded the mail page, then he receives a new mail, then he clicks 'check messages'. the unseen message count does not get updated.

**Comment by Gomez** [comment]
Thanks @tahaalibra! Works now. 

When i create further drafts, the counter does not reflect this immediately. A complete reload is needed. I will create a issue for it. 

The "unseen count for the inbox folder" is a known issue. There is a hacky (closed) PR #1047. Will create a further issue (as we dont have one).

**Comment by Gomez** [comment]
@ChristophWurst And thanks for the explanation! Greatly appreciated!

**Pull request merged**